`CompactionInterruptTest` may cause JVM crush

In rare cases `CompactionInterruptTest` causes JVM to crash. Here is an example: 

Initial Run: https://github.com/hiero-ledger/hiero-consensus-node/actions/runs/23221984940/attempts/1

Build Scan Report: https://scans.gradle.com/s/zvons2vvmigao

Error context:
```
<merkledb: Compacting #1> INFO DataFileCompactor [mergeWhileSnapshotting_ObjectKeyToPath] Finished compaction 10 files / 205.57 MB in 5 ms
<merkledb: Compacting #0> INFO DataFileCompactor [mergeWhileSnapshotting_IdToHashChunk] Finished compaction 10 files / 443.07 KB in 5 ms
<merkledb: Compacting #2> INFO DataFileCompactor [mergeWhileSnapshotting_PathToKeyValue] Finished compaction 10 files / 1.24 GB in 5 ms
pool-2-thread-1 INFO MerkleDbDataSource Closing Data Source [mergeWhileSnapshotting]
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x000076925396b6a8, pid=803, tid=899
#
# JRE version: OpenJDK Runtime Environment Temurin-25.0.2+10 (25.0.2+10) (build 25.0.2+10-LTS)
# Java VM: OpenJDK 64-Bit Server VM Temurin-25.0.2+10 (25.0.2+10-LTS, mixed mode, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
# Problematic frame:
# v  ~StubRoutines::jlong_disjoint_arraycopy 0x000076925396b6a8
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/lib/systemd/systemd-coredump %P %u %g %s %t %c %h %e" (or dumping to /home/runner/_work/hiero-consensus-node/hiero-consensus-node/platform-sdk/swirlds-merkledb/core.803)
#
# An error report file with more information is saved as:
# /home/runner/_work/hiero-consensus-node/hiero-consensus-node/platform-sdk/swirlds-merkledb/hs_err_pid803.log
Warning: [46.123s][warning][os] Loading hsdis library failed
#
# If you would like to submit a bug report, please visit:
#   https://github.com/adoptium/adoptium-support/issues
#

> Task :swirlds-merkledb:test

  0 passing (47.1s)
  1 pending


> Task :swirlds-merkledb:test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':swirlds-merkledb:test'.
> Test process encountered an unexpected problem.
   > Process 'Gradle Test Executor 1' finished with non-zero exit value 134 (this value may indicate that the process was terminated with the SIGABRT signal)

* Try:
> Check common problems https://docs.gradle.org/9.3.1/userguide/java_testing.html#sec:java_testing_troubleshooting.
> Run with --stacktrace option to get the stack trace.
gradle/actions: Writing build results to /home/runner/_work/_temp/.gradle-actions/build-results/gradle-test-1773798147344.json
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org/.

BUILD FAILED in 1m 44s
419 actionable tasks: 180 executed, 239 from cache

Publishing Build Scan to Develocity...
gradle/actions: Writing build results to /home/runner/_work/_temp/.gradle-actions/build-scans/gradle-test-1773798147344.json
https://gradle.com/s/zvons2vvmigao
```

--- a/platform-sdk/swirlds-merkledb/src/test/java/com/swirlds/merkledb/CompactionInterruptTest.java
+++ b/platform-sdk/swirlds-merkledb/src/test/java/com/swirlds/merkledb/CompactionInterruptTest.java
@@ -164,9 +164,9 @@ class CompactionInterruptTest {
                     keyToPathCompactor,
                     pathToKeyValueCompactor);
         } finally {
-            dataSource.close();
             exec.shutdown();
             assertTrue(exec.awaitTermination(10, TimeUnit.SECONDS), "Should not timeout");
+            dataSource.close();
         }
         return true;
     }

